Add DocumentAIProcessorVersion types, mappers, CRD and fuzzer

gemmahou opened the pull request:
### Change description

<!--
Describe what this pull request does.

* If your pull request is to address an open issue, indicate it by specifying the
issue number: 

For example: "Fixes #858"
-->
Fixes #

### Tests you have done

<!--

Make sure you have run "make ready-pr" to run required tests and ensure this PR is ready to review. 

Also if possible, share a bit more on the tests you have done. 

For example if you have updated the pubsubtopic sample, you can share the test logs from running the test case locally.

go test -v -tags=integration ./config/tests/samples/create -test.run TestAll -run-tests pubsubtopic

-->

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

yuwenma commented on apis/documentai/v1alpha1/processor_types.go:
Suggest removing the other resource (processor) and have each PR doing a single task

gemmahou replied:
Jingyi had a [PR](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3557) for processor, I'll remove the files from this PR.

gemmahou replied:
Actually, it's necessary to have processor types, identity and reference because Version has ProcessorRef and Processor as parent... Jingyi's PR is not merged yet. I feel it's fine to make an exception here since Version is a child resource. Ideally we should have parent resource ready before doing child resource.

gemmahou replied:
Actually, I still need processor's reference file, because version has ProcessorRef and processor as its parent. Jingyi's PR is not merged yet, and we may not want to duplicate his PR. I think we can make an exception this time, and include only necessity processor files in this PR.

yuwenma commented on apis/documentai/v1alpha1/processorversion_types.go:
```diff
@@ -0,0 +1,143 @@
+
+	// The KMS key name used for encryption.
+	// +optional
+	KMSKeyName *string `json:"kmsKeyName,omitempty"`
```
This should be a resource ref

gemmahou replied:
Done

yuwenma commented on apis/documentai/v1alpha1/processorversion_types.go:
```diff
@@ -0,0 +1,143 @@
+
+// DocumentAIProcessorVersionSpec defines the desired state of DocumentAIProcessorVersion
+// +kcc:proto=google.cloud.documentai.v1beta3.ProcessorVersion
+type DocumentAIProcessorVersionSpec struct {
```
Can we move the parent to another struct?

gemmahou replied:
This is DocumentAIProcessorVersionSpec struct, when you say parent, do you mean DocumentAIProcessorSpec struct?

yuwenma replied:
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/api-conventions/validations.md#rule-3-parent

yuwenma commented on apis/documentai/v1alpha1/processorversion_types.go:
```diff
@@ -0,0 +1,147 @@
+	// Format:
+	// `projects/{project}/locations/{location}/processors/{processor}/processorVersions/{processor_version}`
+	// +optional
+	Name *string `json:"name,omitempty"`
```
This field is a duplication of metadata.name. We don't want it

gemmahou commented on apis/documentai/v1alpha1/processorversion_types.go:
```diff
@@ -0,0 +1,147 @@
+
+	// The KMS key version with which data is encrypted.
+	// +optional
+	KMSKeyVersionName *string `json:"kmsKeyVersionName,omitempty"`
```
This is a resource reference but the resource is not supported in KCC yet. I migrated kms refs to the new pattern in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3836, also supported KMSKeyVersionRef there. Once that PR merged I'll update this.

yuwenma replied:
Then we should change it to a reference https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/api-conventions/resource-reference.md#basic-rules

yuwenma commented:
/approve

yuwenma commented:
Is this ready for another review?

gemmahou commented:
> Is this ready for another review?

Ideally I'd like to have #3836 first so I can update the KMSKeyRef as well. But to unblock this PR I added KMSKeyVersionRef(alpha) and I could improve the KMSKey Refs in 3836 later. Also removed `name` field as suggested. So this PR is ready now.

yuwenma reviewed (commented):
/lgtm

yuwenma commented on pkg/test/fuzz/generate.go:
```diff
@@ -157,7 +157,7 @@ func fillWithRandom0(t *testing.T, randStream *rand.Rand, msg protoreflect.Messa
 		case protoreflect.DoubleKind:
 			msg.Set(field, protoreflect.ValueOfFloat64(randStream.NormFloat64()))
 		case protoreflect.FloatKind:
-			msg.Set(field, protoreflect.ValueOfFloat64(randStream.NormFloat64()))
```
Mind elaborating why making this change? Can you share the problem this solves?
